roles/README.adoc

```diff
@@ -113,6 +113,8 @@ See also <<vars-vs-defaults,Vars vs Defaults>> which mandates "Avoid embedding l
 Since distribution-specific values are kind of "magic values", it applies to them.
 The same logic applies for providers: a role can load a provider-specific variable file, include a provider-specific task file, or both, as needed.
 Consider making paths to templates internal variables if you need different templates for different distributions.
+Do not use roles as programming interfaces, in the case a common role is necessary to contain common code available to other roles within the same collection, it should not contain user-facing plugins.
+Keep roles loosely coupled with other external roles and variables.
```

Comment:
I think you need to be more precise. Actually it sounds to me like a good heading for other more concrete rules we've already stated.

Reply:
Reworked this based on conversation at https://github.com/redhat-cop/automation-good-practices/pull/68/files#r932600169